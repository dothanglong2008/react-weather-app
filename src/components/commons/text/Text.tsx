import styles from "./Text.module.scss";

type Content = string;

interface TextProps {
  content: Content;
}

function Text({ content }: TextProps) {
  return <span className={styles.text}>{content}</span>;
}

export default Text;
