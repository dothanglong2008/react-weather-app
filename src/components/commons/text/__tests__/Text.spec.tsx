import Text from "../Text";
import renderer from "react-test-renderer";
import { cleanup, render, screen } from "@testing-library/react";

type Container = HTMLDivElement | null;

let container: Container = null;

beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(cleanup);

it("renders Hello text", () => {
  render(<Text content="Hello" />);
  const text = screen.getByText(/Hello/i);
  expect(text).toBeInTheDocument();
});

it("match snapshot with empty text", () => {
  const component = renderer.create(<Text content="" />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders a piece of text with content 'Hello'", () => {
  const content = "Hello";
  const component = renderer.create(<Text content={content} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
