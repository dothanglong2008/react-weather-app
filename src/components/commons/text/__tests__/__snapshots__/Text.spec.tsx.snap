// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot with empty text 1`] = `
<span
  className="text"
/>
`;

exports[`renders a piece of text with content 'Hello' 1`] = `
<span
  className="text"
>
  Hello
</span>
`;
